packages/studio/src/components/PreviewWithUseComponents.tsx

```diff
@@ -19,11 +25,23 @@ export default function PreviewWithUseComponents() {
   const [iframeEl, setIframeEl] = useState<HTMLIFrameElement | null>(null);
   void useImportedComponents(componentTree);
 
+  const dimensions =
+    viewportDimensions.width * (previewRef.current?.clientHeight ?? 0) >
+    viewportDimensions.height * (previewRef.current?.clientWidth ?? 0)
+      ? " w-full "
+      : " h-full ";
+  const id = viewportDimensions.name.replace(/\s+/g, "").toLowerCase();
+  const iframeCSS = twMerge(
+    "mr-auto ml-auto border-2",
+    EXPANDED_VIEWPORTS[id].css,
+    dimensions
+  );
+
```

Comment:
I was thinking we could move this logic and also the `<div ref={previewRef} className="grow w-1/3 bg-white border-8 shadow">` div into IFramePortal?